src/IOManagement.cpp

```diff
@@ -1,9 +1,33 @@
 #include "IOManagement.h"
+#include "adc.h"
+
+volatile Digital_Data digital_data = {0, 0, 0, 0, 0, 0, 0};
+volatile float regen_brake = 0.0f;
 
 void initIO() {
+    // Initialize digital pins
+    pinMode(DIRECTION_SWITCH_PIN, INPUT);
+    pinMode(LEFT_BLINK_PIN, INPUT);
+    pinMode(RIGHT_BLINK_PIN, INPUT);
+    pinMode(CRZ_MODE_A_PIN, INPUT);
+    pinMode(CRZ_SET_PIN, INPUT);
+    pinMode(CRZ_RESET_PIN, INPUT);
+    pinMode(HORN_PIN, INPUT);
 
+    // Initialize analog pins
+    initADC(ADC1);
 }
 
 void readIO() {
-    
+    // Read digital inputs
+    digital_data.direction_switch = digitalRead(DIRECTION_SWITCH_PIN);
+    digital_data.left_blink = digitalRead(LEFT_BLINK_PIN);
+    digital_data.right_blink = digitalRead(RIGHT_BLINK_PIN);
+    digital_data.crz_mode_a = digitalRead(CRZ_MODE_A_PIN);
+    digital_data.crz_set = digitalRead(CRZ_SET_PIN);
+    digital_data.crz_reset = digitalRead(CRZ_RESET_PIN);
+    digital_data.horn = digitalRead(HORN_PIN);
+
+    // Read analog input
+    regen_brake = readADC(REGEN_BRAKE_PIN);
```

Comment:
The input to `readADC()` has to be a channel, not a pin. See [IOManagement.cpp in PDC](https://github.com/badgerloop-software/sc2-pdc/blob/6acd64856e55459d1273929dd11f6d2cfcbd8ad0/src/IOManagement.cpp#L41) for an example.